Initial implementation of a SharedPreferences plugin

ToxicBakery opened the pull request:
#55 initial implementation
- Plugin allows editing for all types excluding Set<String>
- Currently only allows editing of the default shared preferences.

This could probably use some feedback for improvement but it should be a decent foundation.

**Obvious Missing Features**
- Editing `Set<String>`
- Adding new preferences
- Removing preferences
- Viewing other preferences than the default application instance

ToxicBakery committed:
Initial implementation of a SharedPreferences plugin
- Plugin allows editing for all types excluding Set<String>
- Currently only allows editing of the default shared preferences.

ToxicBakery commented:
The plugin also needs its own icon but without guidelines, I was going to leave that to you to decide.

Kritarie commented on hyperion-shared-preferences/src/main/java/com/willowtreeapps/hyperion/sharedpreferences/SharedPreferencesViewerActivity.java:
```diff
@@ -0,0 +1,15 @@
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+public class SharedPreferencesViewerActivity extends AppCompatActivity {
```
Probably want to add `@HyperionIgnore` to this Activity

ToxicBakery replied:
Ah, I must have missed that when I was reviewing the other plugins.

Kritarie reviewed (approved):
Looks good!

ToxicBakery committed:
Added missing ignore to preference viewer activity

Kritarie commented:
I will work on getting an asset for the icon. Only other things I can think of is including a `Toolbar` in the UI with back nav + preferences name. Also, we may want to give the Activity an explicit theme so it doesn't inherit the host application's theme

ToxicBakery committed:
Update to resolve PR issues
- Preference view activity sets a specific theme
- Added toolbar and home back arrow
- Created a theme and colors for the plugin
